tests/test_metadata_schema.py

```diff
@@ -49,13 +49,19 @@ def test_metadata_schema() -> None:
     assert not meta.has_key("g2")
     assert meta.get_key_type("g2") is None
 
-    meta.delete_key("g")
+    meta.delete_key("g", hard=True)
     assert meta.has_key("g")
     assert meta.is_key_deleted("g")
     assert meta.get_key_type("g") is None
+    with pytest.raises(MetadataSchemaError):
```

Comment:
Can we assert on the body of this and the below's error as I believe that they are raising errors for two different reasons.
Can we also split this test into more than one test?